lib/sqlalchemy/orm/strategies.py

```diff
@@ -251,15 +251,18 @@ def setup_query(
         if loadopt and "expression" in loadopt.local_opts:
             columns = [loadopt.local_opts["expression"]]
 
-            for c in columns:
-                if adapter:
-                    c = adapter.columns[c]
-                column_collection.append(c)
+        else:
+            columns = self.parent_property.columns
 
-            fetch = columns[0]
+        for c in columns:
             if adapter:
-                fetch = adapter.columns[fetch]
-            memoized_populators[self.parent_property] = fetch
+                c = adapter.columns[c]
+            column_collection.append(c)
+
+        fetch = columns[0]
```

Comment:
does this include the query_expression() unconditonally?   it should not include if there's no expression, that is, it's hardcoded to NULL.    so some kind of extra state should be added to the property to indicate "load_on_default", something like that.

Reply:
Here it uses expression's type to determine whether the expression should be included. If all expressions are null type expressions, they will be ignored.

Reply:
what do you mean by "state"?  which property of a column hosts its states?

Reply:
"state" here means an instnace variable, like a boolean, "self._should_load_when_not_specified", essentially